corehq/apps/userreports/expressions/specs.py

```diff
@@ -304,3 +305,31 @@ def _get_forms(self, case_id, context):
 
         context.set_cache_value(cache_key, xforms)
         return xforms
+
+
+class SubcasesExpressionSpec(JsonObject):
+    type = TypeProperty('get_subcases')
+    case_id_expression = DefaultProperty(required=True)
+
+    def configure(self, case_id_expression):
+        self._case_id_expression = case_id_expression
+
+    def __call__(self, item, context=None):
+        case_id = self._case_id_expression(item, context)
+        if not case_id:
+            return []
+
+        assert context.root_doc['domain']
```

Comment:
(not a blocker) I thought we moved away from using asserts in favor of raising exceptions?